### ISSUE TITLE
NullPointerExceptions still occur

### ISSUE THREAD

**Issue by nedroden** [Team-Aeros/Vancouver#15]
The problem: Parser.java checks if a HashMap key exists. If that's not the case, it "adds" the element to the array. As one would expect (okay, not really), adding the element does not add the element, causing a NullPointerException to occur. The error occurs only one or twice every 400 clusters, but it's annoying nonetheless.

And yes, the element adding method is synchronized.

Output:
```
:: Vancouver - Weather Data Parser Software
Connection opened on port: 53200
Started thread number: 0
Started thread number: 1
Started thread number: 2
...
Started thread number: 147
Started thread number: 148
Started thread number: 149
Started thread number: 150
Exception in thread "Thread-63" java.lang.NullPointerException
	at com.aeros.controllers.Parser.run(Parser.java:76)
	at com.aeros.controllers.Connection.run(Connection.java:42)
	at java.lang.Thread.run(Thread.java:748)
Exception in thread "Thread-76" java.lang.NullPointerException
	at com.aeros.controllers.Parser.run(Parser.java:78)
	at com.aeros.controllers.Connection.run(Connection.java:42)
	at java.lang.Thread.run(Thread.java:748)
Started thread number: 151
Started thread number: 152
Started thread number: 153
Started thread number: 154
...
Started thread number: 390
Started thread number: 391
Started thread number: 392
Started thread number: 393
Started thread number: 394
Started thread number: 395
Started thread number: 396
Started thread number: 397
Started thread number: 398
Started thread number: 399
```

**Comment by nedroden** [comment]
Fixed in #14

**Comment by nedroden** [comment]
Haven't encountered any NullPointerExceptions since this was 'fixed', so closing this issue.
